src/strands/agent/agent.py

```diff
@@ -617,11 +614,24 @@ def _record_tool_execution(
             tool: The tool call information.
             tool_result: The result returned by the tool.
             user_message_override: Optional custom message to include.
-            messages: The message history to append to.
         """
         # Create user message describing the tool call
+
+        # Filter out non-serializable objects to prevent JSON serialization errors
```

Comment:
I don't like this approach of serializing everything one by one. Instead we can use `default`. See https://stackoverflow.com/a/56138540

Reply:
That's a good callout